FirebaseAdmin/FirebaseAdmin/Auth/UserRecord.cs

```diff
@@ -24,20 +25,82 @@ namespace FirebaseAdmin.Auth
     /// of this class are immutable and thread safe.
     /// </summary>
-    internal class UserRecord
+    public sealed class UserRecord : IUserInfo
     {
+        private const string PROVIDERID = "firebase";
+
         private string uid;
+        private string email;
+        private string phoneNumber;
+        private bool emailVerified;
+        private string displayName;
+        private string photoUrl;
+        private bool disabled;
+        private List<ProviderUserInfo> providers;
+        private long tokensValidAfterTimestamp;
+        private UserMetadata userMetaData;
         private IReadOnlyDictionary<string, object> customClaims;
 
-        public UserRecord(string uid)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UserRecord"/> class with the specified user ID.
+        /// </summary>
+        /// <param name="uid">The user's ID.</param>
+        internal UserRecord(string uid)
         {
-            this.Uid = uid;
+            if (string.IsNullOrEmpty(uid))
+            {
+                throw new ArgumentNullException(nameof(uid));
+            }
+
+            this.uid = uid;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UserRecord"/> class from an existing instance of the <see cref="GetAccountInfoResponse.User"/> class.
+        /// </summary>
+        /// <param name="user">The <see cref="GetAccountInfoResponse.User"/> instance to copy the user's data from.</param>
+        internal UserRecord(GetAccountInfoResponse.User user)
+        {
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+            else if (string.IsNullOrEmpty(user.UserID))
+            {
+                throw new ArgumentException("UserID must not be null or empty.");
+            }
+
+            this.uid = user.UserID;
+            this.email = user.Email;
+            this.phoneNumber = user.PhoneNumber;
+            this.emailVerified = user.EmailVerified;
+            this.displayName = user.DisplayName;
+            this.photoUrl = user.PhotoUrl;
+            this.disabled = user.Disabled;
+
+            if (user.Providers == null || user.Providers.Count == 0)
+            {
+                this.providers = new List<ProviderUserInfo>();
+            }
+            else
+            {
+                var count = user.Providers.Count;
+                this.providers = new List<ProviderUserInfo>(count);
+
+                for (int i = 0; i < count; i++)
+                {
+                    this.providers.Add(new ProviderUserInfo(user.Providers[i]));
+                }
+            }
+
+            this.tokensValidAfterTimestamp = user.ValidSince * 1000;
+            this.userMetaData = new UserMetadata(user.CreatedAt, user.LastLoginAt);
+            this.customClaims = UserRecord.ParseCustomClaims(user.CustomClaims);
```

Comment:
This doesn't seem to be exposed at the moment.

Reply:
I'm not sure I understand your comment on this one.

Reply:
We need a public property that exposes this attribute. Please disregard if you already have that, and I just missed it somehow.

Reply:
Yeah, those properties do exist:
`
/// <summary>
/// Gets a timestamp representing the date and time that this token will become active.
/// </summary>
public long TokensValidAfterTimestamp => this.tokensValidAfterTimestamp;

/// <summary>
/// Gets additional user metadata.
/// </summary>
public UserMetadata UserMetaData => this.userMetaData;

/// <summary>
/// Gets or sets the custom claims set on this user.
/// </summary>
[JsonIgnore]
public IReadOnlyDictionary<string, object> CustomClaims
{
	get => this.customClaims;
	set
	{
		CheckCustomClaims(value);
		this.customClaims = value;
	}
}
`
Assuming that's what you meant, I'll go ahead and mark this conversation as resolved then.